cpp/daal/src/services/compiler/generic/env_detect_features.cpp

```diff
@@ -166,6 +166,45 @@ static int check_avx512_features()
     return 1;
 }
 
+static int check_avx512_vnni_features()
+{
+    /*
+    CPUID.(EAX=07H, ECX=0H):ECX.AVX512_VNNI[bit 11]==1
+    */
+    uint32_t avx512_vnni_mask = (1 << 11);
+    if (!check_cpuid(7, 0, 2, avx512_vnni_mask))
```

Comment:
CPUID checks can be done in more compact form. For example, the same functionality in [recent SVS CPUArch dispatcher PR](https://github.com/intel/ScalableVectorSearch/pull/113/files#diff-003c4dcf7f1013f31e10a67ce5b3ae354f2e5b4784d8b56d1a2f7f7b8e601d5d).

Reply:
Thanks to pointing to the approach. I don't think we need this variety of checks in oneDAL, but I'll try to adapt this tabular approach.